game-core/src/main/java/games/strategy/engine/framework/startup/ui/panels/main/game/selector/GameSelectorModel.java

```diff
@@ -237,6 +239,9 @@ private static boolean gameUriExistsOnFileSystem(final String gameUri) {
    * method will find the location of the zip file itself.
    */
   private static Path getDefaultGameRealPath(final URI defaultGame) {
+    // The file system of the URI needs to be created before Path.of can be called.
+    // So, first see if the file system is already created and if that throws
+    // FileSystemNotFoundException, then try and create it.
```

Comment:
Is there a significant cost to creating the file system?

Reply:
I don't think there is a significant cost.  I can't find anything online easily.  But I believe the code that later opens the zip also has to create the filesystem internally.